### PR TITLE
Add popup delay option

### LINKED ISSUE #115: Allow user to set delay before menu pops up
It would be useful if users could set an optional delay that would prevent the menu from popping up until they stopped typing for a specified number of milliseconds; if typing quickly enough, the menu wouldn't pop up until you paused typing briefly. This would also help prevent thrashing/flickering, as the menu would be called less often.

I thought this is what `let g:asyncomplete_completion_delay`, but that doesn't seem to be the case.

### PR DESCRIPTION
**Pull request by jsit**
Closes #115

### CODE REVIEW

File: autoload/asyncomplete.vim [hunk 1]
```diff
@@ -447,7 +447,7 @@ function! s:python_refresh_completions(ctx) abort
         endif
     endfor
 
-    call s:core_complete(a:ctx, l:startcol, l:filtered_matches, s:matches)
+    call timer_start(g:asyncomplete_popup_delay, {timer->s:core_complete(a:ctx, l:startcol, l:filtered_matches, s:matches)})
```
Comment (prabirshrestha) [inline autoload/asyncomplete.vim]: If `0` can we not directly call `s:core_complete` without `timer_start`?

Comment (jsit) [reply autoload/asyncomplete.vim]: Hm I don't know the best way to do this; if we do `timer_start` with `0` ms, it just happens immediately anyway. Is it worth adding an extra conditional?

Comment (prabirshrestha) [reply autoload/asyncomplete.vim]: I'm worried about this https://github.com/vim/vim/issues/1713#issuecomment-455795951.

Comment (jsit) [reply autoload/asyncomplete.vim]: Good point -- fixed.

**Comment by prabirshrestha** [comment]
@jsit Can you send the PR for v2 branch instead. v2 branch is now usable but has few bugs.

**Comment by jsit** [comment]
Replaced by #121 for `v2` branch.

**Pull request closed without merging**